File: Ej49/C++/arreglo.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

int encontrarMenor(int vect[], int tam){
	int menor=vect[0];
	for (int i = 1; i < tam; i++){
		if (vect[i] < menor) {
			menor = vect[i];
		}
	}
	return menor;
}

int main(){
	int tam = 4;
	int vect[tam];

	cout << "Valores de arreglo: " << endl;

	srand(time(NULL));	
	for (int i = 0; i < tam; i++){
		vect[i] = rand() % 101;
		cout << "Valor " << i << ": " << vect[i] << endl;
	}
	cout << endl;
	
	cout << "Valor menor: " << encontrarMenor(vect, 4) << endl;
	return 0;
}

File: Ej49/Java/Arreglo.java
import java.util.Scanner;
import java.util.Random;

class Arreglo {
	public static void main(String [] args){
		int tam = 4;
		int[] vect = new int[tam];
		Random rnd = new Random();
		System.out.println("Valores de arreglo: ");

		for (int i = 0; i < tam; i++){
			vect[i] = rnd.nextInt(101);
			System.out.println("Valor " + i + ": " + vect[i]);
		}

		System.out.print("\n");

		System.out.println("Valor menor: " + encontrarMenor(vect, tam));
	}

	public static int encontrarMenor(int vect[], int tam){
		int menor = vect[0];
		for (int i = 1; i < tam; i++){
			if (vect[i] < menor){
				menor = vect[i];
			}
		}
		return menor;
	}
}